README.md
Indexed Search Add Markers
==========================

With the help of this extension the administrator is able to add custom markers to EXT:indexed_search.

This is necessary if a website has several languages, but the administrator want's to avoid having several templates just to localize some strings.

The current status of the extension is a very low alpha, it does not do very much yet. At this point of time it just adds custom markers to a row.

Usage
-----

The following example TypoScript is adding the marker relevance to a row of the search result (dont forget to add ###RELEVANCE### to your template file) and modifies the title-marker.

    ### Additional markers for indexed search
    plugin.tx_indexedsearchaddmarkers {
        rowView {
            markers {
                relevance = TEXT
                relevance {
                    value = Relevance
                    lang.de = Relevanz
                    lang.fr = Pertinence
                    lang.cz = Relevance
                    wrap = |:
                }

                #title = TEXT
                #title {
                #	field = item_title
                #	wrap = <em>|</em>
                #}
            }
        }
    }

License
-------

GNU General Public License version 2

The GNU General Public License can be found at http://www.gnu.org/copyleft/gpl.html.

Author
------

Dan Untenzu (<untenzu@webit.de>)
for webit! Gesellschaft für neue Medien mbH (http://www.webit.de/)

Changelog
---------

./Changelog.md

Contribution
------------

> TYPO3 - inspiring people to share!

This TYPO3 extension is Open Source, so please use, patch, extend or fork it.


ChangeLog
2010-07-23 Dan Untenzu  <untenzu@webit.de>

	* initial code generated with kickstarter


src/typo3/indexedsearch_addmarkers/hooks/class.tx_indexedsearchaddmarkers_indexedsearch_prepareResultRowTemplateDataHook.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Dan Untenzu <untenzu@webit.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */

class tx_indexedsearchaddmarkers_indexedsearch_prepareResultRowTemplateDataHook extends tslib_pibase {

	/**
	 * Add some custom markers to indexed_search - here: additional markers for each row of the search result
	 *
	 * @param array &$hookParams	Parameters give by the source extension -> currently 'markerArray' with all already existing markers and 'event' with all fields of the current row
	 * @return array	$tmplContent Marker array
	 * @access public
	 */
	function prepareResultRowTemplateData_postProc($tmplContent, $row, $headerOnly) {
		$this->cObj = t3lib_div::makeInstance('tslib_cObj'); // local cObj.

		$this->conf = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_indexedsearchaddmarkers.'];

		$this->conf['view'] = 'row';
		// pass row to TypoScript (use as »field = item_title« e.g.)
		$this->cObj->data = $row;

		$this->getCustomMarkers($tmplContent);

		return $tmplContent;
	}

	function getCustomMarkers(&$markerArray) {
		// build custom markers in TypoScript ->  plugin.tx_indexedsearchaddmarkers.row.markers.[markername] = TEXT
		foreach($this->conf[$this->conf['view'] . 'View.']['markers.'] as $marker => $name) {
			// only use cObject name (foo), not config (foo.)
			if(!strstr($marker, '.')) {
				// replace marker ###FOO### with type given in $name (.markers.foo = TEXT) and config indentified by $marker (.markers.foo.value = bar)
				// 2010-07-23 -> dont use ###-scheme, since indexed_search converts the array a while later
				//$markerArray['###' . strtoupper($marker) . '###'] = $this->cObj->cObjGetSingle(
				//	$name, $this->conf[$this->conf['view'] . 'View.']['markers.'][$marker . '.']
				//);
				$markerArray[$marker] = $this->cObj->cObjGetSingle(
					$name, $this->conf[$this->conf['view'] . 'View.']['markers.'][$marker . '.']
				);
			}
		}
	}
}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/indexedsearch_addmarkers/hooks/class.tx_indexedsearchaddmarkers_indexedsearch_prepareResultRowTemplateDataHook.php'])    {
    include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/indexedsearch_addmarkers/hooks/class.tx_indexedsearchaddmarkers_indexedsearch_prepareResultRowTemplateDataHook.php']);
}
?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "indexedsearch_addmarkers".
#
# Auto generated 23-07-2010 18:59
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Add markers to indexed_search',
	'description' => 'Add some custom markers to indexed_search search results (necessary if a website has several languages, but the administrator want\'s to avoid having several templates just to localize some strings).',
	'category' => 'fe',
	'author' => 'Dan Untenzu',
	'author_email' => 'untenzu@webit.de',
	'shy' => '',
	'dependencies' => 'indexed_search',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'alpha',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => '',
	'version' => '1.0.0',
	'constraints' => array(
		'depends' => array(
			'typo3' => '4.5.0-6.2.99',
			'indexed_search' => '',
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
);

?>